mgmt/src/grpc/server.rs

```diff
@@ -0,0 +1,1617 @@
+                        // Parse CIDR into IP and netmask
+                        let (ip_str, netmask) = self.parse_cidr(cidr)?;
+                        // Add as an include rule for AS
+                        vpc_expose = vpc_expose.as_range(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                    gateway_config::config::peering_as::Rule::Not(not) => {
+                        // Parse CIDR into IP and netmask for exclude rule
+                        let (ip_str, netmask) = self.parse_cidr(not)?;
+                        // Add as an exclude rule for AS
+                        vpc_expose = vpc_expose.not_as(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                }
+            } else {
+                return Err("PeeringAs must have either cidr or not field set".to_string());
+            }
+        }
+
+        Ok(vpc_expose)
+    }
+
+    /// Parse a CIDR string into IP and netmask
+    fn parse_cidr(&self, cidr: &str) -> Result<(String, u8), String> {
+        let parts: Vec<&str> = cidr.split('/').collect();
+        if parts.len() != 2 {
+            return Err(format!("Invalid CIDR format: {}", cidr));
+        }
+
+        let ip = parts[0].to_string();
+        let netmask = parts[1]
+            .parse::<u8>()
+            .map_err(|_| format!("Invalid netmask in CIDR {}: {}", cidr, parts[1]))?;
+
+        Ok((ip, netmask))
+    }
+
+    /// Convert Overlay from gRPC
+    fn convert_overlay_from_grpc(
+        &self,
+        overlay: &gateway_config::Overlay,
+    ) -> Result<Overlay, String> {
+        // Create VPC table
+        let mut vpc_table = VpcTable::new();
+
+        // Add VPCs
+        for vpc_grpc in &overlay.vpcs {
+            // Convert VPC
+            let vpc = self.convert_vpc_from_grpc(vpc_grpc)?;
+
+            // Add to table - use proper method pattern
+            vpc_table
+                .add(vpc)
+                .map_err(|e| format!("Failed to add VPC {}: {}", vpc_grpc.name, e))?;
+        }
+
+        // Create peering table
+        let mut peering_table = VpcPeeringTable::new();
+
+        // Add peerings
+        for peering_grpc in &overlay.peerings {
+            // Convert peering
+            let peering = self.convert_peering_from_grpc(peering_grpc)?;
+
+            // Add to table
+            peering_table
+                .add(peering)
+                .map_err(|e| format!("Failed to add peering {}: {}", peering_grpc.name, e))?;
+        }
+
+        // Create overlay with the tables
+        Ok(Overlay::new(vpc_table, peering_table))
+    }
+
+    /// Convert DeviceConfig to gRPC Device
+    pub fn convert_device_to_grpc(
+        &self,
+        dev: &DeviceConfig,
+    ) -> Result<gateway_config::Device, String> {
+        let driver = match dev.settings.driver {
+            PacketDriver::Kernel(_) => 0, // Kernel
+            PacketDriver::DPDK(_) => 1,   // DPDK
```

Comment:
```suggestion
            PacketDriver::Kernel(_) => 0,
            PacketDriver::DPDK(_) => 1,
```

We can probably figure these out from the enum field names :slightly_smiling_face: